lua/net.lua

```diff
@@ -13,6 +17,20 @@ function M.setup_block()
 	conf.num_rx_queues = 3
 	conf.num_tx_queues = 3
 
+	conf.front.num_ports = #front_ports
+	conf.front.name = "front"
+	conf.front.pci_addrs = ffi.new("const char *[" .. #front_ports .. "]")
```

Comment:
Maybe there's an easier way than the solution I came up with, but I ran into the issue that in order to pass the strings from Lua to C, they have to be declared as `const char **`, so you can't later re-use that part of `struct net_config` to do dynamic allocation and store the strings permanently for C. So I ended up using `struct net_config` to both pass constant parameters from Lua (with a note saying not to use them) and also hold the dynamically allocated versions. from